File: study_group/2202/42840.py
def solution(answers):
    answer = []
    score = [0,0,0]
    stu1 = [1,2,3,4,5]
    stu2 = [2,1,2,3,2,4,2,5]
    stu3 = [3,3,1,1,2,2,4,4,5,5]
    index = 0
    max = 0
    for a in answers:
        if a == stu1[index%len(stu1)]:
            score[0] += 1
        if a == stu2[index%len(stu2)]:
            score[1] += 1
        if a == stu3[index%len(stu3)]:
            score[2] += 1
        index += 1
    max = sorted(score)[-1]
    for x in range(3):
        if max == score[x]:
            answer.append(x+1)
    return answer

test_case = [1,2,3,4,5]
# test_case = [1,3,2,4,2]
print(solution(test_case))

File: study_group/2202/42840.js
function solution(answers) {
    const stu1 = [1,2,3,4,5]
    const stu2 = [2,1,2,3,2,4,2,5]
    const stu3 = [3,3,1,1,2,2,4,4,5,5]
    let score = [0,0,0]
    let result = []
    let max = 0
    for (let index=0; index<answers.length; index++){
        if (answers[index] == stu1[index%(stu1).length]){
            score[0]++;
        }
        if (answers[index] == stu2[index%(stu2).length]){
            score[1]++;
        }
        if (answers[index] == stu3[index%(stu3).length]){
            score[2]++;
        }
    }
    for (let x = 0; x<3; x++){
        if (max < score[x]){
            max = score[x]
        }
    }
    for (let x = 0; x<3; x++){
        if (max == score[x]){
            result.push(x+1)
        }
    }
    return result
}

// let test_case = [1,2,3,4,5];
let test_case = [1,3,2,4,2];
// let test_case = [3, 3, 2, 1, 5];
console.log(solution(test_case));